Assets/LeapMotionModules/InteractionEngine/Scripts/ControllerImplementations/HeuristicGrabClassifier.cs

```diff
@@ -0,0 +1,142 @@
+﻿using UnityEngine;
+using System;
+using System.Collections.Generic;
+using Leap.Unity.RuntimeGizmos;
+
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+
+namespace Leap.Unity.Interaction {
+  public class HeuristicGrabClassifier {
+    Collider[] collidingCandidates = new Collider[10];
+    public InteractionManager _manager;
+    private Hand _hand;
```

Comment:
Be very careful when keeping references to hands around, they can change out from underneath you because we pool them on the provider side.  If you need a reference to the entire hand, you should create a new hand and copy the data over.